### PR TITLE
Set required and default properties for kibana auth.

### PR DESCRIPTION
**Pull request by jmcarp**
At the moment, kibana auth configuration variables are passed as arbitrary hashes to `kibana.env`. As my team noticed, that means it's easy to specify an invalid configuration by forgetting required properties. There's also no good way to set default properties. This patch adds properties for the `kibana-auth-plugin` package so that we can make some properties required and set defaults for others. I'm sending a second patch to `logsearch-boshrelease` that allow the kibana control script to source arbitrary files.

I was also thinking about handling this using bosh links and exposing properties from this job to the kibana job in `logsearch-boshrelease`, but I slightly prefer this approach because it allows us to decouple the bosh property names from the environment variables that the auth plugin expects.

cc @cnelson @rogeruiz @wjwoodson

### CODE REVIEW

File: jobs/kibana-auth-plugin/config/config.sh.erb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export USE_HTTPS="<%= p("kibana-auth.use_https") %>"
+export CF_API_URL="<%= p("kibana-auth.cloudfoundry.api_endpoint") %>"
+export KIBANA_OAUTH2_CLIENT_ID="<%= p("kibana-auth.cloudfoundry.skip_ssl_validation") %>"
+export KIBANA_OAUTH2_CLIENT_SECRET="<%= p("kibana-auth.cloudfoundry.client_id") %>"
```
Comment (rogeruiz) [inline jobs/kibana-auth-plugin/config/config.sh.erb]: Looks like these properties need to be corrected and I don't see the skip SSL validation property being used here.

**Review by rogeruiz — CHANGES_REQUESTED** [review]
Looks good besides the typo. Elegant solution! 👍🏽

**Review by Infra-Red — CHANGES_REQUESTED** [review]
@jmcarp Could you also update deployment templates with new properties?

**Comment by jmcarp** [comment]
Thanks for reviewing @Infra-Red. I updated the templates.

**Review by Infra-Red — APPROVED** [review]

**Pull request merged**